vm/src/protocol/number.rs

```diff
@@ -0,0 +1,291 @@
+use std::borrow::Cow;
+
+use crate::{
+    builtins::{int, PyByteArray, PyBytes, PyComplex, PyFloat, PyInt, PyIntRef, PyStr},
+    common::lock::OnceCell,
+    function::ArgBytesLike,
+    stdlib::warnings,
+    AsObject, PyObject, PyPayload, PyRef, PyResult, TryFromBorrowedObject, VirtualMachine,
+};
+
+#[allow(clippy::type_complexity)]
+#[derive(Clone)]
+pub struct PyNumberMethods {
+    /* Number implementations must check *both*
+    arguments for proper type and implement the necessary conversions
+    in the slot functions themselves. */
```

Comment:
Does this mean we need to check the first argument(&PyNumber) is a correct PyNumber object?
In this case, I feel like the type will fit better in `&PyObject` rather than `&PyNumber`

Reply:
The comment is from cpython but as the source shows that cpython only does the check for binary operations which could be called invertly but assume the type is correct for other functions